pkg/engine/policyContext.go

```diff
@@ -151,25 +138,26 @@ func (c *PolicyContext) WithOldResource(resource unstructured.Unstructured) *Pol
 	return copy
 }
 
-func (c *PolicyContext) WithResources(newResource unstructured.Unstructured, oldResource unstructured.Unstructured) *PolicyContext {
-	return c.WithNewResource(newResource).WithOldResource(oldResource)
-}
-
-func (c *PolicyContext) withAdmissionOperation(admissionOperation bool) *PolicyContext {
+func (c *PolicyContext) WithResourceKind(gvk schema.GroupVersionKind, subresource string) *PolicyContext {
```

Comment:
Is `WithResourceKind(..)` set wherever we use `policyContext`? https://github.com/kyverno/kyverno/pull/6544 had green CI which turned red after this merge.

On investigation, `WithResourceKind(..)` was not set for the `policyContext` which eventually led to `matchesResourceDescription(..)` giving an error. I will fix it for that particular use case in the same PR, but we need to check this throughout the codebase.

Reply:
If not set we take apiVersion and kind from the resource.

Reply:
But that would be a problem for subresources. They can have kinds like `PodExecOptions` and `Binding` which will be checked against policy's kind which is not supported.

Reply:
Yes that’s why it should be set by the caller.

Reply:
It might be less error prone to make it mandatory.

Reply:
Yes, making it mandatory is better, if the caller wants to use `resource.GVK` then they can use it.